packages/language-server/lib/language-services.js

```diff
@@ -1,8 +1,11 @@
 const { DiagnosticSeverity, Range } = require("vscode-languageserver");
 const { map } = require("lodash");
 const { parse } = require("@xml-tools/parser");
+const { buildAst } = require("@xml-tools/ast");
+const { checkConstraints } = require("@xml-tools/constraints");
 
 const SYNTAX_ERROR_MSG = "Syntax error";
+const XML_WELL_FORMEDNESS = "well-formedness";
```

Comment:
I am trying to align with the specifications.
- https://www.w3.org/TR/xml/#sec-well-formed

Not sure what is the best choice.

Reply:
Not sure the user will understand what this means and what is the difference between well-formedness and syntax errors.

Reply:
But on the other hand it is not exactly a syntax error and the end user can google it and reach this result:
- https://www.w3resource.com/xml/well-formed.php

Reply:
I googled `xml "well-formedness"` and didn't find either of these links. The wikipedia article mentions things we report as syntax errors (and none of the things being validated here). This might be a problem with wikipedia or the exact search terms the user searched with, but it means it might not be so easy to find information about, and it's such a technical detail that I don't think it has to be displayed to the user (unless we add an option to turn off these checks specifically).
It's not a critical issue, just might be confusing for users.

Reply:
When you use quotes google searches for an exact term.
both `xml well-formdness` and `well-formdness xml` showed the W3S very easy to understand page as the first result.

So while the term is highly technical, it could help for some users.
But TBH I now think, perhaps we should display the name of the extension or the language server there?

Reply:
What do we do in UI5-Langauge-Assistant?

Reply:
The extension name sounds good since VSCode describe this property as: `A human-readable string describing the source of this diagnostic, e.g. 'typescript' or 'super lint'.`

Reply:
Yeah we seem to have mis-used this property.
I will change this in a separate PR (for both sources).